scripts/run_simulation.py

```
@@ -1440,5 +1504,265 @@ def plot_results(  # noqa: C901, PLR0912, PLR0913, PLR0915
+
+    # Save weights per agent
+    weights_dir = data_dir / "weights"
+    weights_dir.mkdir(parents=True, exist_ok=True)
+    for agent in agents:
+        if isinstance(agent.brain, WeightPersistence):
+            weight_file = (
+                "final.pt" if agent.agent_id == "default" else f"final_{agent.agent_id}.pt"
+            )
+            try:
+                save_weights(agent.brain, weights_dir / weight_file)
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)